src/Controller/LoginController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace ItkDev\OpenIdConnectBundle\Controller;
+
+use ItkDev\OpenIdConnect\Security\OpenIdConfigurationProvider;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class LoginController extends AbstractController
+{
+
+    /**
+     * @var array
+     */
+    private $openIdProviderOptions;
+
+    /**
+     * @var string
+     */
+    private $returnRoute;
+
+    public function __construct(array $openIdProviderOptions, string $returnRoute)
+    {
+        $this->openIdProviderOptions = $openIdProviderOptions;
+        $this->returnRoute = $returnRoute;
+    }
+
+    /**
+     * @Route("/login", name="itk_dev_openid_connect_login")
```

Comment:
You could probably remove this Route annotation as the route is configured in the bundles routes configuration.